MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,344 @@
+
+In v1 of the SDK, the default was to send request to /oauth/token using json, allowing to opt-in to use x-www-form-urlencoded by setting the `useFormData` flag to _true_.
+
+With v2, we have flipped the default value for `useFormData` to **true**, meaning we will be sending requests to Auth0’s token endpoint using `application/x-www-form-urlencoded` as the content type by default.
+
+> :warning: This can affect existing rules and actions, and it’s important to ensure all your actions still work as expected after upgrading to v2.
+> To restore the original behaviour, you can set `useFormData` to **false**, and your rules and actions should continue to work as before.
+
+### No more iframe fallback by default when using refresh tokens
+
+When using refresh tokens in v1, we fall back to using iframes whenever a refresh token exchange would fail. This has caused problems before in environments that do not support iframes, and we have specifically introduced `useRefreshTokensFallback` to be able to opt-out of falling back to iframes in the case a refresh_grant fails.
+
+With v2, we have flipped the default value for `useRefreshTokensFallback` to false, meaning, by default, we will not be falling back to using iframes when `useRefreshTokens` is set to true and exchanging the refresh token fails.
+
+If you would want to restore the original behaviour, and still fallback to iframes when the refresh_grant fails, you can set `useRefreshTokensFallback` to true.
+
+### `getUser` and `getIdTokenClaims`
+
+With v1 of our SDK, both `getUser` and `getIdTokenClaims` support providing an (optional) audience and scope to retrieve the user information.
+
+```ts
+const user = await getUser();
+const user = await getUser({ audience, scope });
+
+const claims = await getIdTokenClaims();
+const claims = await getIdTokenClaims({ audience, scope });
+```
+
+As an application should only have one user, it makes little sense to be passing these parameters when trying to retrieve the current user.
+
+With v2, both `getUser` and `getIdTokenClaims` have dropped support for any arguments, and should be called as such.
+
+```ts
+const user = await getUser();
+const claims = await getIdTokenClaims();
+```
+
+### Changes to default scopes (profile and email)
+
+Our SDK defaults to requesting `openid profile email` as the scopes. However, when the user explicitly sets the `scope` when constructing `Auth0Client`, v1 would still include `openid profile email` as well.
+
+With v2, we have reworked this to still default to `openid profile email` when the scope property has been omitted, but only include openid when the user sets a scope explicitly.
+
+This means that the following code in v1:
+
+```ts
+const client = new Auth0Client({
+  scope: 'scope1'
+});
+```
+
+Needs to be updated to explicitly include profile email to achieve the same in v2:
+
+```ts
+const client = new Auth0Client({
+  scope: 'profile email scope1'
+});
+```
+
+#### defaultScope
```

Comment:
I would explicitly clarify here as well that `advancedOptions` no longer exists (if that's the case, I'm assuming it is). People might not realise that it only had one option in it.